lib/jsdom/level2/html.js

```diff
@@ -2048,6 +2048,17 @@ function loadFrame (frame) {
   }
 }
 
+function refreshAccessors(document) {
+  var window = document.defaultView;
+  var frames = Array.prototype.slice.call(document.querySelectorAll("iframe,frame"));
+  frames.forEach(function (frame, i) {
```

Comment:
Two things:

- Use `Array.prototype.forEach.call(frames` instead of slicing-then-forEaching, for efficiency.
- Perhaps we should set `window._length = frames.length` here and omit all other manual manipulation of `_length` below?